### ISSUE TITLE
Array type inference issue

### ISSUE THREAD

**Issue by robryanx** [SerenityOS/jakt#1436]
```
mut row: [bool] = [false, false, false, false, false, false, false]
mut board: [[bool]] = []

board.push(row[..].to_array())
```
This gives `Type mismatch: expected ‘[T]’, but got ‘[bool]’`

Using a temp variable works:
```
let current_row = row[..].to_array()
board.push(current_row)
```

**Comment by cybergsus** [comment]
Just tested my fix for #1246 and this issue looked like it was failing because of incorrect type substitution, so I tested it manually, and it works! Should I put your MWE under `tests/typechecker` as well?

**Comment by robryanx** [comment]
Sure, would make sense to have a test for this.
